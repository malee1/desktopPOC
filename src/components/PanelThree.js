import { React, useState } from 'react';
import { makeStyles } from '@material-ui/core/styles';
import { Box, Typography, List, ListItem, Button, Collapse } from '@material-ui/core';
import { Notes, Announcement, Notifications, CheckBox, ArrowDropDown } from '@material-ui/icons';
import { useSelector } from 'react-redux';
import { panelThreeLabels } from '../constants/panelLabels';
import DropDownPanel from './DropDownPanel';

const useStyles = makeStyles({
  root: {
    flexGrow: 1,
    padding: '5px'
  },
  box: {
    width: '27vw'
  },
  list: {
    flexDirection: 'column',
    whiteSpace: 'nowrap'
  },
  listItem: {
    backgroundColor: 'white'
  },
  listItemButton: {
    justifyContent: 'flex-end',
    backgroundColor: 'white',
    display: 'flex'
  },
  iconContainer: {
    width: '8%',
    backgroundColor: 'white'
  },
  icon: {
    backgroundColor: 'white'
  },
  labelContainer: {
    width: '40%',
    marginLeft: '2px'
  },
  label: {
    backgroundColor: 'white',
    paddingLeft: '5px'
  },
  labelValueContainer: {
    width: '55%',
    marginLeft: '2px'
  },
  labelValue: {
    backgroundColor: 'white',
    paddingLeft: '5px'
  },
  dropDown: {
    // backgroundColor: 'grey',
    zIndex: '1000',
    position: 'absolute',
    height: '15vh',
    width: '27vw'
  },
  panelButton: {
    height: '30px'
  }
});

export default function PanelOne() {
  const classes = useStyles();

  const labels = panelThreeLabels;

  const customer = useSelector((state) => state.customer.data);

  const [open, setOpen] = useState(false);

  const handleDropDown = () => {
    setOpen(!open);
  };

  return (
    <div className={classes.root}>
      <Box className={classes.box} bgcolor="primary.main">
        <List className={classes.list}>
          <ListItem className={classes.listItem}>
            <div className={classes.iconContainer}>
              <Notifications className={classes.icon} />
            </div>
            <div className={classes.labelContainer}>
              <Typography className={classes.label} variant="body1">
                {labels.alerts}
              </Typography>
            </div>
            <div className={classes.labelValueContainer}>
              <Typography className={classes.labelValue} variant="body1">
                {customer.alerts}
              </Typography>
            </div>
          </ListItem>
          <ListItem className={classes.listItem}>
            <div className={classes.iconContainer}>
              <Announcement className={classes.icon} />
            </div>
            <div className={classes.labelContainer}>
              <Typography className={classes.label} variant="body1">
                {labels.service}
              </Typography>
            </div>
            <div className={classes.labelValueContainer}>
              <Typography className={classes.labelValue} variant="body1">
                {customer.service}
              </Typography>
            </div>
          </ListItem>
          <ListItem className={classes.listItem}>
            <div className={classes.iconContainer}>
              <Notes className={classes.icon} />
            </div>
            <div className={classes.labelContainer}>
              <Typography className={classes.label} variant="body1">
                {labels.keyInfo}
              </Typography>
            </div>
            <div className={classes.labelValueContainer}>
              <Typography className={classes.labelValue} variant="body1">
                {customer.keyInfo}
              </Typography>
            </div>
          </ListItem>
          <ListItem className={classes.listItem}>
            <div className={classes.iconContainer}>
              <CheckBox className={classes.icon} />
            </div>
            <div className={classes.labelContainer}>
              <Typography className={classes.label} variant="body1">
                {labels.auth}
              </Typography>
            </div>
            <div className={classes.labelValueContainer}>
              <Typography className={classes.labelValue} variant="body1">
                {customer.auth}
              </Typography>
            </div>
          </ListItem>
          <ListItem className={classes.listItemButton}>
            <Button
              className={classes.panelButton}
              onClick={handleDropDown}
              variant="contained"
              color="secondary"
            >
              More Alert Details
              <ArrowDropDown />
            </Button>
          </ListItem>
        </List>
        <Collapse in={open} timeout="auto" unmountOnExit>
          <div className={classes.dropDown}>
            <DropDownPanel />
          </div>
        </Collapse>
      </Box>
    </div>
  );
}
