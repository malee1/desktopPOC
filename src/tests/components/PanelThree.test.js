import React from 'react';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import renderer from 'react-test-renderer';
import { cleanup, render, fireEvent } from '@testing-library/react';

import PanelThree from '../../components/PanelThree';

const mockStore = configureMockStore();
const store = mockStore({
  customer: {
    data: {
      name: 'Test Test',
      dob: '1111',
      uRef: '2222',
      tel: '3333',
      timeWait: '10 mins',
      context: 'test test',
      recording: 'Inactive',
      callInfo: 'ABCDE',
      alerts: 'aaaa',
      service: 'bbbb',
      keyInfo: 'cccc',
      auth: 'dddd'
    }
  }
});

afterEach(() => {
  cleanup();
  jest.clearAllMocks();
});

describe('the panel renders', () => {
  it('matches the snapshot', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <PanelThree />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('has one component', () => {
    const { getAllByTestId } = render(
      <Provider store={store}>
        <PanelThree />
      </Provider>
    );
    const container = getAllByTestId('panel-three');
    expect(container.length).toBe(1);
  });

  it('has four items', () => {
    const { getAllByTestId } = render(
      <Provider store={store}>
        <PanelThree />
      </Provider>
    );
    const container = getAllByTestId('item');
    expect(container.length).toBe(4);
  });
});

describe('it shows the correct content', () => {
  it('has the correct name', () => {
    const { getAllByText } = render(
      <Provider store={store}>
        <PanelThree />
      </Provider>
    );
    // const container = getAllByText('Test Test');
    expect(getAllByText('aaaa').length).toBe(1);
    expect(getAllByText('bbbb').length).toBe(1);
    expect(getAllByText('cccc').length).toBe(1);
    expect(getAllByText('dddd').length).toBe(1);
  });
});

describe('button click works', () => {
  it('on click the function is called', () => {
    const { getByText, getAllByTestId } = render(
      <Provider store={store}>
        <PanelThree />
      </Provider>
    );
    const btn = getByText('More Alert Details');
    fireEvent.click(btn);
    const dropPanel = getAllByTestId('drop-down-panel');
    expect(dropPanel.length).toBe(1);
  });
});
