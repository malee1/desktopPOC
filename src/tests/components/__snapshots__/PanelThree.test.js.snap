// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the panel renders matches the snapshot 1`] = `
<div
  className="makeStyles-root-1"
  data-testid="panel-three"
>
  <div
    className="MuiBox-root MuiBox-root-14 makeStyles-box-2"
  >
    <ul
      className="MuiList-root makeStyles-list-3 MuiList-padding"
    >
      <li
        className="MuiListItem-root makeStyles-listItem-4 MuiListItem-gutters"
        data-testid="item"
        disabled={false}
      >
        <div
          className="makeStyles-iconContainer-6"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root makeStyles-icon-7"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 22c1.1 0 2-.9 2-2h-4c0 1.1.89 2 2 2zm6-6v-5c0-3.07-1.64-5.64-4.5-6.32V4c0-.83-.67-1.5-1.5-1.5s-1.5.67-1.5 1.5v.68C7.63 5.36 6 7.92 6 11v5l-2 2v1h16v-1l-2-2z"
            />
          </svg>
        </div>
        <div
          className="makeStyles-labelContainer-8"
        >
          <p
            className="MuiTypography-root makeStyles-label-9 MuiTypography-body1"
          >
            Customer Alerts: 
          </p>
        </div>
        <div
          className="makeStyles-labelValueContainer-10"
        >
          <p
            className="MuiTypography-root makeStyles-labelValue-11 MuiTypography-body1"
          >
            aaaa
          </p>
        </div>
      </li>
      <li
        className="MuiListItem-root makeStyles-listItem-4 MuiListItem-gutters"
        data-testid="item"
        disabled={false}
      >
        <div
          className="makeStyles-iconContainer-6"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root makeStyles-icon-7"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M20 2H4c-1.1 0-1.99.9-1.99 2L2 22l4-4h14c1.1 0 2-.9 2-2V4c0-1.1-.9-2-2-2zm-7 9h-2V5h2v6zm0 4h-2v-2h2v2z"
            />
          </svg>
        </div>
        <div
          className="makeStyles-labelContainer-8"
        >
          <p
            className="MuiTypography-root makeStyles-label-9 MuiTypography-body1"
          >
            Service Alerts: 
          </p>
        </div>
        <div
          className="makeStyles-labelValueContainer-10"
        >
          <p
            className="MuiTypography-root makeStyles-labelValue-11 MuiTypography-body1"
          >
            bbbb
          </p>
        </div>
      </li>
      <li
        className="MuiListItem-root makeStyles-listItem-4 MuiListItem-gutters"
        data-testid="item"
        disabled={false}
      >
        <div
          className="makeStyles-iconContainer-6"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root makeStyles-icon-7"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M3 18h12v-2H3v2zM3 6v2h18V6H3zm0 7h18v-2H3v2z"
            />
          </svg>
        </div>
        <div
          className="makeStyles-labelContainer-8"
        >
          <p
            className="MuiTypography-root makeStyles-label-9 MuiTypography-body1"
          >
            Key Info: 
          </p>
        </div>
        <div
          className="makeStyles-labelValueContainer-10"
        >
          <p
            className="MuiTypography-root makeStyles-labelValue-11 MuiTypography-body1"
          >
            cccc
          </p>
        </div>
      </li>
      <li
        className="MuiListItem-root makeStyles-listItem-4 MuiListItem-gutters"
        data-testid="item"
        disabled={false}
      >
        <div
          className="makeStyles-iconContainer-6"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root makeStyles-icon-7"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
            />
          </svg>
        </div>
        <div
          className="makeStyles-labelContainer-8"
        >
          <p
            className="MuiTypography-root makeStyles-label-9 MuiTypography-body1"
          >
            Authorisation
          </p>
        </div>
        <div
          className="makeStyles-labelValueContainer-10"
        >
          <p
            className="MuiTypography-root makeStyles-labelValue-11 MuiTypography-body1"
          >
            dddd
          </p>
        </div>
      </li>
      <li
        className="MuiListItem-root makeStyles-listItemButton-5 MuiListItem-gutters"
        disabled={false}
      >
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-panelButton-13 MuiButton-containedSecondary"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            More Alert Details
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
          </span>
        </button>
      </li>
    </ul>
  </div>
</div>
`;
